db/prefixeddb/prefixeddb.go

```diff
@@ -0,0 +1,124 @@
+package prefixeddb
+
+import (
+	"bytes"
+
+	"go.vocdoni.io/dvote/db"
+)
+
+// PrefixedDatabase wraps a db.Database prefixing all keys with `prefix`.
+type PrefixedDatabase struct {
+	prefix []byte
+	db     db.Database
+}
+
+// check that PrefixedDatabase implements the db.Database interface
+var _ db.Database = (*PrefixedDatabase)(nil)
+
+// NewPrefixedDatabase creates a new PrefixedDatabase.  If the db is already a
+// PrefixedDatabase, instead of wrapping again, the prefixes are appended to
+// avoid unecessay layers.
+func NewPrefixedDatabase(db db.Database, prefix []byte) *PrefixedDatabase {
+	if pdb, ok := db.(*PrefixedDatabase); ok {
+		return &PrefixedDatabase{append(pdb.prefix, prefix...), pdb.db}
+	}
+	return &PrefixedDatabase{prefix, db}
+}
+
+// Notice that this method doesn't close the wrapped db.Database.
+func (d *PrefixedDatabase) Close() error {
+	return nil
+}
+
+// ReadTx returns a db.ReadTx
+func (d *PrefixedDatabase) ReadTx() db.ReadTx {
+	return NewPrefixedReadTx(d.db.ReadTx(), d.prefix)
+}
+
+// WriteTx returns a db.WriteTx
+func (d *PrefixedDatabase) WriteTx() db.WriteTx {
+	return NewPrefixedWriteTx(d.db.WriteTx(), d.prefix)
+}
+
+// Iterate implements the db.Database.Iterate interface method
+func (d *PrefixedDatabase) Iterate(prefix []byte, callback func(key, value []byte) bool) error {
+	return d.db.Iterate(append(d.prefix, prefix...), func(key, value []byte) bool {
```

Comment:
note that upstream badger does support iterating over a prefix natively. I guess we lose that ability given that our DB interface is too high level. maybe leave a TODO here to support badgerdb shortcuts in the future?

Reply:
The Iterate method as implemented for badgerdb uses the native prefix: https://github.com/vocdoni/vocdoni-node/blob/698b3434f1381c656e3907ac60f35461da7aa3a0/db/badgerdb/badger.go#L131
So we are already using the prefix Iteration of badgerdb here.  Am I missing something?

Reply:
Ah, you're right, I forgot this.